### PR TITLE
Iow 461

### PR DESCRIPTION
**Pull request by ssoper-usgs**
Before making a pull request
----------------------------

- [ ] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	execute format('create table if not exists %I.time_series_grades_%s partition of %I.time_series_grades for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+	execute format('create table if not exists %I.time_series_header_info_%s partition of %I.time_series_header_info for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+	execute format('create table if not exists %I.time_series_interpolation_types_%s partition of %I.time_series_interpolation_types for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+	execute format('create table if not exists %I.time_series_methods_%s partition of %I.time_series_methods for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+	execute format('create table if not exists %I.time_series_qualifiers_%s partition of %I.time_series_qualifiers for values in (%L)', '${AQTS_SCHEMA_NAME}', month_name, '${AQTS_SCHEMA_NAME}', month_number);
+
+	/* recreate the indexes */
```
Comment (dsteinich) [inline liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql:32]: We need to recreate all of the indexes/constraints on all of the tables.

**Review by dsteinich — CHANGES_REQUESTED** [review]
I think the wqp pattern for partitions is better - they are declared with the table, rather than a separate partition folder - it is ok the define the changeset the way you have, just merge them into the individual table's folder. (https://github.com/NWQMC/schema-wqp-core/tree/postgres/liquibase/changeLogs/wqp/schemaLoad/tables/actMetric)

File: liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql (hunk 1 above)
Comment (ssoper-usgs) [reply liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql:32]: Yes. When we're ready to merge let's discuss strategy.

Comment (ssoper-usgs) [reply liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql:32]: I see what you're saying now.  We had a discussion and resolved the issue.  Also added partitions/constraints/indexes for the time_series_points table.

**Review by dsteinich — CHANGES_REQUESTED** [review]
The old indexes need to be carried forward to the partition tables. This is in addition to manually creating the unique indexes that the primary key and unique constraints created behind the scenes for us.

File: liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql [hunk 2]
```diff
@@ -0,0 +1,70 @@
+create or replace function ${AQTS_SCHEMA_NAME}.prune_time_series_data(date timestamp)
```
Comment (dsteinich) [inline liquibase/changeLogs/capture/functions/pruneTimeSeriesData/pruneTimeSeriesData.sql]: I think we should working with a date here, not a timestamp. Date might be a reserved word, if so the name of the parameter should be something else. At the very least, line 1 here needs to match line 16 in the changelog ('character varying' vs 'timestamp')

**Review by dsteinich — CHANGES_REQUESTED** [review]

**Pull request merged**